flocker/node/test/test_deploy.py

```diff
@@ -1308,6 +1311,572 @@ def test_discover_manifestation_paths(self):
             self.successResultOf(d).paths)
 
 
+class ApplicationNodeDeployerDiscoverNodeConfigurationTests(
+        SynchronousTestCase):
+    """
+    Tests for ``ApplicationNodeDeployer.discover_local_state``.
+    """
+    def setUp(self):
+        self.network = make_memory_network()
+
+    def test_discover_none(self):
+        """
+        ``ApplicationNodeDeployer.discover_local_state`` returns an empty
+        ``NodeState`` if there are no Docker containers on the host.
+        """
+        fake_docker = FakeDockerClient(units={})
+        api = ApplicationNodeDeployer(
+            u'example.com',
+            docker_client=fake_docker,
+            network=self.network
+        )
+        d = api.discover_local_state(NodeState(hostname=u'example.com'))
+
+        self.assertEqual(NodeState(hostname=u'example.com',
+                                   manifestations=None,
+                                   paths=None),
+                         self.successResultOf(d))
+
+    def test_discover_one(self):
+        """
+        ``ApplicationNodeDeployer.discover_local_state`` returns ``NodeState``
+        with a a list of running ``Application``\ s; one for each active
+        container.
+        """
+        expected_application_name = u'site-example.com'
+        unit = Unit(name=expected_application_name,
+                    container_name=expected_application_name,
+                    container_image=u"flocker/wordpress:latest",
+                    activation_state=u'active')
+        fake_docker = FakeDockerClient(units={expected_application_name: unit})
+        application = Application(
+            name=unit.name,
+            image=DockerImage.from_string(unit.container_image)
+        )
```

Comment:
I ended up making corresponding shared `Application` instance, so that reduced duplication.